docs/_advanced/configuring_workflow_steps.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Configuring and updating workflow steps
+lang: en
+slug: configuring-steps
+order: 10
+beta: true
+---
+{% raw %} 
+<div class='section-content'>
+⚠️ Workflow [steps from apps](https://api.slack.com/workflows/steps) is a beta feature. As the feature is developed, **Bolt for JavaScript's API will change to add better native support.** To develop with workflow steps in Bolt, use the `@slack/bolt@feat-workflow-steps` version of the package rather than the standard `@slack/bolt`.
+
+When a builder is adding your step to a new or existing workflow, your app will need to configure and update that step:
+
+**1. Listening for `workflow_step_edit` action**
+
+When a builder initially adds (or later edits) your step in their workflow, your app will receive a `workflow_step_edit` action. Your app can listen to `workflow_step_edit` using `action()` and the `callback_id` in your app's configuration.
+
+**2. Opening and listening to configuration modal**
+
+The `workflow_step_edit` action will contain a `trigger_id` which your app will use to call `views.open` to open a modal of type `workflow_step`. This configuration modal has more restrictions than typical modals—most notably you cannot include `title`, `submit`, or `close` properties in the payload.
+
+To learn more about configuration modals, [read the documentation](https://api.slack.com/workflows/steps#handle_config_view).
+
+Similar to other modals, your app can listen to this `view_submission` payload with the built-in [`views()` method](#view_submissions).
+
+**3. Updating the builder's workflow**
+
+After your app listens to the `view_submission`, you'll call [`workflows.updateStep`](https://api.slack.com/methods/workflows.updateStep) with the unique `workflow_step_id` (found in the `body`'s `workflow_step` object) to save the configuration for that builder's specific workflow. Two important parameters:
+- `inputs` is an object with keyed child objects representing the data your app expects to receive from the user upon workflow step execution. You can include handlebar-style syntax (`{{ variable }}`) for variables that are collected earlier in a workflow.
+- `outputs` is an array of objects indicating the data your app will provide upon workflow step completion.
+
+Read the documentation for [`input` objects](https://api.slack.com/reference/workflows/workflow_step#input) and [`output` objects](https://api.slack.com/reference/workflows/workflow_step#output) to learn more about how to structure these parameters.
+
+</div>
+{% endraw %} 
+
+```javascript
+// Your app will be called when user adds your step to their workflow
+app.action({ type: 'workflow_step_edit', callback_id: 'add_task' }, async ({ body, ack, client }) => {
+  // Acknowledge the event
+  ack();
```

Comment:
@shaydewael I'm sorry for my late review here but `await ack();` is more appropriate here.